sqlglot/dialects/tsql.py

```diff
@@ -580,6 +585,16 @@ def _parse_if(self) -> t.Optional[exp.Expression]:
 
             return super()._parse_if()
 
+        def _parse_period_for_system_time(self) -> exp.PeriodForSystemTimeConstraint:
+            self._match(TokenType.TIMESTAMP_SNAPSHOT)
+            self._match_l_paren()
+
+            start = self._parse_id_var()
+            self._match(TokenType.COMMA)
+            end = self._parse_id_var()
+            self._match_r_paren()
```

Comment:
Ditto about moving this whole method in the base class. We can also replace these lines with:

```python
self._match(TokenType.TIMESTAMP_SNAPSHOT)
id_vars = self._parse_wrapped_id_vars
return exp.PeriodForSystemTimeConstraint(start=seq_get(id_vars, 0), end=seq_get(id_vars, 1))
```